llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -433,6 +433,13 @@ struct AAAMDAttributesFunction : public AAAMDAttributes {
       indicatePessimisticFixpoint();
       return;
     }
+
+    for (Instruction &I : instructions(F)) {
+      if (isa<AllocaInst>(I) || isa<AddrSpaceCastInst>(I)) {
```

Comment:
only addrspacecasts from address space 5 require flat_scr. Just an alloca does not imply flat_scr is required, only if the alloca is ever cast to flat. 


I suppose for robustness you could double check the alloca address space is correct

Reply:
Approximately yes, but you should use dyn_cast and never static_cast

Reply:
> Approximately yes, but you should use dyn_cast and never static_cast

Doesn't the check `isa<AddrSpaceCastInst>(I)` already make sure that the instruction is a `AddrSpaceCastInst`?

Reply:
@arsenm ping.

Reply:
``static_cast`` is a C++ feature, but rarely used in LLVM code. Instead we use ``dyn_cast`` or ``cast``. You should be doing something like this:

`if (auto *AddrCastInst = dyn_cast<AddrSpaceCastInst>(&I)) { ... }`

Reply:
Thanks. This has been updated.